airflow/providers/google/cloud/hooks/dataprep.py

```diff
@@ -50,26 +54,60 @@ def _headers(self) -> Dict[str, str]:
         }
         return headers
 
-    @property
-    def _token(self) -> str:
-        conn = self.get_connection(self.dataprep_conn_id)
-        token = conn.extra_dejson.get("token")
-        if token is None:
-            raise AirflowException(
-                "Dataprep token is missing or has invalid format. "
-                "Please make sure that Dataprep token is added to the Airflow Connections."
-            )
-        return token
-
     @retry(stop=stop_after_attempt(5), wait=wait_exponential(multiplier=1, max=10))
     def get_jobs_for_job_group(self, job_id: int) -> Dict[str, Any]:
         """
         Get information about the batch jobs within a Cloud Dataprep job.
 
-        :param job_id The ID of the job that will be fetched.
+        :param job_id The ID of the job that will be fetched
         :type job_id: int
         """
-        url: str = f"{self._url}/{job_id}/jobs"
+
+        endpoint_path = f"/v4/jobGroups/{job_id}/jobs"
+        url: str = os.path.join(self._base_url, endpoint_path)
```

Comment:
Using `path.join` requires no leading slash
```suggestion
        endpoint_path = f"v4/jobGroups/{job_id}/jobs"
        url: str = os.path.join(self._base_url, endpoint_path)
```
because:
```
In [2]: os.path.join("https://app:123/path/", "/api/v1/")
Out[2]: '/api/v1/'

In [3]: os.path.join("https://app:123/path/", "api/v1/")
Out[3]: 'https://app:123/path/api/v1/'
```
And indeed in logs I see
```
requests.exceptions.MissingSchema: Invalid URL '/v4/jobGroups': No schema supplied.
```

Reply:
Done.